App.js

```diff
@@ -2,16 +2,36 @@ import React from 'react';
 import { StyleSheet, Text, View, WebView } from 'react-native';
 import { registerForPushNotificationsAsync } from './lib/pushNotifications';
 
+const loggedInUrlRegex = /members/;
+const mainUrl = 'https://www.timeoverflow.org/';
+
 export default class App extends React.Component {
-  componentDidMount() {
-    registerForPushNotificationsAsync();
+  componentDidMount() {}
+
+  registerForPushNotifications() {
+    const token = await registerForPushNotificationsAsync();
+
+    this.runRemoteTokenStoring(token);
+  }
+
+  onNavigationStateChange({ url }) {
```

Comment:
You can do the following, to avoid binding to `this` on the render method. It's magic.

```diff
-  onNavigationStateChange({ url }) { 
+ onNavigationStateChange = ({url}) => {
+ ...
+ }
```

Reply:
wat?! can you explain that? I'm just a poor rubyist trying to understand this...

Reply:
haha, maybe the current syntax is more accessible then :P

Reply:
I shouldn't be the one taken as example :trollface: